### PR TITLE
add openapi spec file for ddl service.

### PR DESCRIPTION
**Pull request by hzhzhang**
PR checklist:
- Did you run ClangFormat ?
- Did you run CheckStyle ?
- Did you check the comment / log / exception conventions in Engineering code process wiki page ?
- Did you import unnecessary headers ?
- Did you surround `Daisy : starts/ends` for new code in existing ClickHouse code base ?

Please write user-readable short description of the changes:

### CODE REVIEW

File: spec/rest-api/ddl-service.yaml [hunk 1]
```diff
@@ -0,0 +1,500 @@
+openapi: 3.0.0
```
Comment (chenziliang) [inline spec/rest-api/ddl-service.yaml:1]: Can we add a REST API to service this file ? Something like `dae/v1/apis` ? For exact API path, could you please dig out Gang's REST API guide ?

File: spec/rest-api/ddl-service.yaml [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+    post:
+      summary: Create a tabular table
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/tabularTable'
+      parameters:
+        - name: x-bdg-request-id
```
Comment (chenziliang) [inline spec/rest-api/ddl-service.yaml:20]: What about `username` ?

File: spec/rest-api/ddl-service.yaml [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+    delete:
+      summary: Drop an existed rawstore table named table_name
+      parameters:
+        - name: table_name
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: x-bdg-request-id
+          in: query
```
Comment (chenziliang) [inline spec/rest-api/ddl-service.yaml]: Is this in `query` or in `head` or both ?

Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml]: It is in query.

Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml]: It is in query.

Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml]: x-bdg-request-id is in query while X-ClickHouse-Query-Id is in header

File: spec/rest-api/ddl-service.yaml (hunk 2 above)
Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml:20]: The following params are added: X-ClickHouse-Query-Id X-ClickHouse-User X-ClickHouse-Key X-ClickHouse-Quota user password and quota_key

File: spec/rest-api/ddl-service.yaml (hunk 1 above)
Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml:1]: Will add in another PR AND a ticket is created.

File: spec/rest-api/ddl-service.yaml (hunk 3 above)
Comment (chenziliang) [reply spec/rest-api/ddl-service.yaml]: `x-bdg-request-id` should be in header as well.

Comment (hzhzhang) [reply spec/rest-api/ddl-service.yaml]: reset it into header

**Review by chenziliang — APPROVED** [review]

**Pull request merged**